classes/i18n/Locale.php

```diff
@@ -353,6 +353,21 @@ public function getScripts(?string $locale = null): Scripts
         return $this->_getLocaleCache(__METHOD__, $locale, fn () => $this->_getIsoCodes($locale)->getScripts());
     }
 
+    /**
+     * Get the locales counts by only locale code
+     */
+    public function getLocaleCodesCount(array $allLocales = []): array
+    {
+        if (empty($allLocales)) {
+            $allLocales = $this->getLocales();
+        }
+        return array_count_values(
+            collect(array_keys($allLocales))
+                ->map(fn(string $value) => trim(explode('_', $value)[0]))
```

Comment:
Have you seen cases where the `trim` is necessary? I'm pretty sure other parts of the code would break if there were spaces in the locale code. (Elsewhere also)

Reply:
Not really . it's more like a habit of mine to do a bit of defensive programming. probably not necessary .

Reply:
I understand, but I'd rather not do this unless it's necessary. For a long time we allowed vague data to be stored in the database -- `0` and `null` and `''` and `[]` were interchangeable, for example. It led to overcomplicated/slow queries and lots of areas where data was acceptable in one part of the system but caused errors in another. But the most challenging thing was managing upgrades where it was necessary to anticipate boundary data conditions in the wild and remember them across several releases. In the last 3 or so years we've been changing our approach towards have clear, unambiguous expectations of data -- `null` is distinct from `[]` and each has a specific meaning; foreign key integrity is enforced in the database; types/nullability in PHP are hinted; etc. The idea is for data errors to surface as soon as possible and for the underlying cause to be fixed, rather than adding work-arounds ad-hoc and incompletely. That way we can catch/fix them quickly during the dev cycle rather than having end users find boundary conditions through variety of data.

Reply:
@asmecher I understand . I have removed the `trim` from `Locale@getFormattedDisplayNames` .